code/modules/mob_spawn/mob_spawn.dm

```diff
@@ -150,42 +153,70 @@
 	if(!SSticker.HasRoundStarted() || !loc)
 		return
 
+	if(ckeys_trying_to_spawn)
+		// We don't open the prompt more than once at a time.
+		if(user.ckey in ckeys_trying_to_spawn)
+			return
+
+	var/user_ckey = user.ckey // Just in case shenanigans happen, we always want to remove it from the list.
+	LAZYADD(ckeys_trying_to_spawn, user_ckey)
+
 	if(prompt_ghost)
 		var/prompt = "Become [prompt_name]?"
 		if(user.can_reenter_corpse && user.mind)
 			prompt += " (Warning, You can no longer be revived!)"
 		var/ghost_role = tgui_alert(usr, prompt, buttons = list("Yes", "No"), timeout = 10 SECONDS)
 		if(ghost_role != "Yes" || !loc || QDELETED(user))
+			ckeys_trying_to_spawn -= user_ckey
```

Comment:
Why not LAZYREMOVE?

Reply:
I'm not using `LAZYREMOVE` since by then we're certain the list will exist, due to the prior `LAZYADD()`. It's be a redundant null check that would never catch anything.

Reply:
We use lazy remove to remove the list when it's empty.